core/domain/stats_services.py

```diff
@@ -84,20 +82,33 @@ def get_top_state_rule_answers(
     listed in 'rule_str_list'. The number of answers returned is the number of
     rule spec strings based in multiplied by top_answer_count_per_rule.
     """
-    answer_logs = stats_domain.StateRuleAnswerLog.get_multi(
-        exploration_id, [{
-            'state_name': state_name,
-            'rule_str': rule_str
-        } for rule_str in rule_str_list])
+    return get_top_state_rule_answers_multi(
+        [(exploration_id, state_name)], rule_str_list,
+        top_answer_count_per_rule)[0]
+
 
-    all_top_answers = []
-    for answer_log in answer_logs:
-        top_answers = answer_log.get_top_answers(top_answer_count_per_rule)
-        all_top_answers += [
-            {'value': top_answer[0], 'count': top_answer[1]}
-            for top_answer in top_answers
-        ]
-    return all_top_answers
+def get_top_state_rule_answers_multi(
+        exploration_state_list, rule_str_list, top_answer_count_per_rule):
+    """Returns a list of top answers (by submission frequency) submitted to the
+    given explorations and states which were mapped to any of the rules listed
+    in 'rule_str_list'. The number of answers returned is at most the number of
+    rule spec strings multiplied by top_answer_count_per_rule.
```

Comment:
Maybe clarify length and order of the return value in relation to the input parameters (similar to above).